### DISCUSSION TITLE
Connecting to MySQL with self-signed certificate

### DISCUSSION THREAD

**Discussion by blambert** [Budibase/budibase#2478] [Help] [answered]
I'm trying Budibase for the first time and in setting up a MySQL datasource, I see that the SSL configuration is a freeform series of keys/values.  I'd like to connect to a server that has a "self-signed" certificate, but cannot figure out the configuration values needed to make this happen.

I have tried "rejectUnauthorized: false" and "sslVerify: false." When I save the configuration and click "Fetch tables from database" I get the "Error: self signed certificate in certificate chain" message in red at the top.

The documentation for the mysql npm package (https://github.com/mysqljs/mysql#ssl-options) shows the rejectUnauthorized option, but it doesn't seem to be working in this case.

I couldn't find any notes, discussions or issues mentioning this, so I'm hoping someone here can help.  Thanks!

**Comment by mjashanks** [comment]
Hi @blambert - I think you should find the options listed here: https://www.npmjs.com/package/mysql#ssl-options

We use the node `mysql` package to connect.

**Comment by shogunpurple** [comment, marked as answer]
Hi @blambert - I think this one could be a result of using later versions of the MySQL server. We need to upgrade our `mysql` connector to the `mysql2` package. You can follow along with the progress of this here:

https://github.com/Budibase/budibase/issues/1899

**Reply by blambert** [reply]
Yes, the server I'm attempting to connect to is running 8.0.19.  Thanks!
